inc/customizer/customizer.php

```diff
@@ -559,7 +559,8 @@ function create_css_builder(){
 			if ( isset( $setting['callback'] ) && isset( $setting['default'] ) && $val != $setting['default'] ) {
 				$val = get_theme_mod($id);
 				if ( $val != $setting['default'] ) {
-					if ( ! is_numeric( $val ) ) {
+					// Reset the value to the default if the setting is meant to be a number
+					if ( is_numeric( $setting['default'] ) && ! is_numeric( $val ) ) {
```

Comment:
Seems like this is sanitization, which should be happening in the `sanitize_callback` functions.

Reply:
Ok, so after digging for a bit it seems the underlying issue here isn't sanitization, but rather that defaults aren't being used by the callback functions. It's only presented an issue for the Site Title font size setting as that's the only one with a non-empty default value.

So the fix here is that either each callback needs to do it's own checking for empty values and apply the default, or we use the built-in mechanism and pass default values in to `get_theme_mod`.

We'd just need to find out why the original code was only calling callbacks when the value was **not** the default.

Reply:
I submitted PR #284 which reverts the general changes and just applies the default for the specific setting. It seems to me that the intention is for callbacks to handle empty values.